tests/test_mariadb.py

```diff
@@ -135,6 +143,38 @@ def test_mariadb_db_env_vars(
             cur.execute("SELECT * FROM test;")
             assert cur.fetchone() == (1, 100, "abc'def")
 
+    if root_pw is None:
+        for line in host.check_output(
+            f"{container_runtime.runner_binary} logs {container_per_test.container_id}",
+        ).splitlines():
+            if "GENERATED ROOT PASSWORD: " in line:
+                root_pw = line.partition("GENERATED ROOT PASSWORD: ")[
+                    2
+                ].strip()
+                break
+
```

Comment:
The connect in the next line will certainly fail if it's `None`, but it'll make the error more readable

```suggestion
    assert root_pw, "Root password must be either set or obtained from the log"
```